### DISCUSSION TITLE
Kafka 1 / Kafka 3 Node Setup?

### DISCUSSION THREAD

**Discussion by stx-hseow** [Seagate/cortx-utils#299] [Q&A]
Looking at the [Kafka Server Setup](https://github.com/Seagate/cortx-utils/wiki/Kafka-Server-Setup), it has both Kafka 1 and Kafka 3 Node setup.

Even though they both download the same RPM, but there are different steps and I don't really understand what does the title mean. Why 2 different setups?

**Comment by hessio** [comment]
Hi @harrison-seow-seagate let me attempt to answer based on my understanding. So, it seems for most configurations/setups of XORTX you can have a single node setup, but you can also have 3 node cluster. This is the R1 version of CORTX in future releases we plan to have this so that you can scale up to N nodes, where N is theoretically infinity. 

So it seems to me that the Kafka server needs to see all 3 nodes in your CORTX cluster and the 3 node setup just has the extra step of adding the 2 extra nodes in the configuration files that exist. 

I think something that might help is if we update this page to have the same setup steps and then just say "do this for 3 nodes and do this for single node setup" at the steps which are different. i will create a PR for this.

Hope this was helpful, if you have anymore questions please feel free to ask.

**Comment by rahul-tripathi-git** [comment]
Hi @harrison-seow-seagate 

There is an option to have 3 node kafka as well as 1 node kafka.

If the cortx is single node, 1 node kafka will be required, If cortx is N node (future plans as rightly mentioned by @hessio), 3 node kafka can be setup for all the benefits 3 node setup can provide.

Yes rpm is same, but in case of 3Node setup, that rpm needs to be installed on all the nodes (for Cortx 3-Node setup) and then further steps needs to be followed as mentioned in 3Node section.

Please feel free to reach out with further queries.
